Libraries/LibJS/Runtime/StringPrototype.cpp

```diff
@@ -307,4 +308,61 @@ Value StringPrototype::trim_end(Interpreter& interpreter)
     return trim_string(interpreter, *this_object, TrimMode::Right);
 }
 
+Value StringPrototype::substring(Interpreter& interpreter)
+{
+    auto* this_object = interpreter.this_value().to_object(interpreter.heap());
```

Comment:
Missing a null check here. `to_object()` may fail due to an exception being thrown.
```suggestion
    auto* this_object = interpreter.this_value().to_object(interpreter.heap());
    if (!this_object)
        return {};
```